### PR TITLE
Deprecating `Shorcuts.of` and `Shortctus.maybeOf`

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This deprecates `Shorcuts.of` and `Shortctus.maybeOf` because they're not especially useful, since the only thing you can really set on a `ShortcutManager` is the shortcuts, and the `Shortcuts` widget that you give it to manages those, so if it rebuilds, it overwrites what you set.

Once this is finally removed, it will also eliminate an `InheritedWidget` for each `Shortcuts` widget.

## Tests
- Added tests for Shortcuts.manager

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: only affects comments (and deprecation annotations)

**Review by goderbauer — APPROVED** [review]
LGTM

As discussed, I think we should also deprecate the manager param on the Shortcuts widget and provide a separate constructor, that only takes a manager, but no list of shortcuts to clean up that API as well.

File: packages/flutter/lib/src/widgets/shortcuts.dart [hunk 1]
```diff
@@ -883,13 +894,22 @@ class Shortcuts extends StatefulWidget {
...
   /// See also:
   ///
   ///  * [of], which is similar to this function, but returns a non-nullable
   ///    result, and will throw an exception if it doesn't find a [Shortcuts]
   ///    ancestor.
+  @Deprecated(
+    'Finding a ShortcutsManager in this way will no longer be supported. '
+    'This feature was deprecated after v3.1.0-0.0.pre.',
+  )
+  // When this is finally removed, _ShortcutsMarker can also be removed.
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/shortcuts.dart]: Done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request closed without merging**